src/app/frontend/common/components/serializedreference/serializedreference_directive.js

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SerializedReferenceController from './serializedreference_controller';
+
+/**
+ * Returns directive definition for serialized references.
+ *
+ * @return {!angular.Directive}
+ */
+export default function serializedReferenceDirective() {
```

Comment:
Can you make this a component? This has been our standard practice so far. Everything is a component, unless it uses some of the directive features. 

There are examples in the codebase to inspire on. Basically, you turn this into an object instead of a function returning an object. Delete some lines and change referencectrl to `$ctrl`.